django/db/models/fields/mixins.py

```diff
@@ -24,3 +26,36 @@ def set_cached_value(self, instance, value):
 
     def delete_cached_value(self, instance):
         del instance._state.fields_cache[self.get_cache_name()]
+
+
+class CheckFieldDefaultMixin:
+    """Check the default value of a field to make sure it's not a mutable object."""
+    _default_hint = ('<valid default>', '<invalid default>')
+
+    def _check_default(self):
+        if self.has_default():
+            if self.default is None or callable(self.default) or isinstance(
+                self.default, (bool, int, float, str, tuple, frozenset)
```

Comment:
to make sure that the default value is not mutable, Should we use a util function like this?
```
def is_immutable(value):
    return value is copy(value)
```
or try to change the value and return the result as the exception.

Reply:
Thanks, I didn't think of that. I'm not sure if the logic is equivalent now, but I can't think of a case where it's not.